### ISSUE TITLE
Add DefenderForStorageSettings resource

### ISSUE THREAD

**Issue by barclayadam** [pulumi/pulumi-azure-native#2996]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

I'm trying to change the Microsoft Defender for Storage settings, but can find no way of doing that via Pulumi Azure native. It's available in the REST API (https://learn.microsoft.com/en-us/azure/storage/common/azure-defender-storage-configure?tabs=enable-subscription#rest-api-1):

PUT
https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/providers/Microsoft.Security/DefenderForStorageSettings/current?api-version=2022-12-01-preview

````json
{
    "properties": {
        "isEnabled": true,
        "malwareScanning": {
            "onUpload": {
                "isEnabled": true,
                "capGBPerMonth": 5000
            }
        },
        "sensitiveDataDiscovery": {
            "isEnabled": true
        },
        "overrideSubscriptionLevelSettings": true
    }
}
````

**Comment by mjeffryes** [comment]
Thanks for the report @barclayadam; I was able to find some examples in the spec for this feature: https://github.com/Azure/azure-rest-api-specs/blob/fa8683f6725e5447cc808420c4542ce8375f0325/specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/examples/DefenderForStorage/GetDefenderForStorageSettings_example.json; we'll have to look into why it doesn't make it into the generated provider.

**Comment by thomas11** [comment]
We cannot automatically include it because the spec has no delete operation for it, so we cannot map it to the complete Pulumi resource lifecycle.

We can include it by adding a suitable default value, if one exists, to `pkg/openapi/defaults/defaultResourcesState.go`. Then, it will be included as a resource and the Pulumi delete operation will reset the value of an instance to the default instead of actually deleting it.

**Comment by thomas11** [comment]
These are the resource properties.
```json
      "properties": {
        "isEnabled": {
          "type": "boolean",
          "description": "Indicates whether Defender for Storage is enabled on this storage account."
        },
        "malwareScanning": {
          "x-ms-client-flatten": true,
          "$ref": "#/definitions/MalwareScanningProperties"
        },
        "sensitiveDataDiscovery": {
          "x-ms-client-flatten": true,
          "$ref": "#/definitions/SensitiveDataDiscoveryProperties"
        },
        "overrideSubscriptionLevelSettings": {
          "type": "boolean",
          "description": "Indicates whether the settings defined for this storage account should override the settings defined for the subscription."
        }
      },
```

Looks like a suitable default would simply be `isEnabled=false`? I'm not familiar with this Azure service, though.

**Comment by barclayadam** [comment]
Thank you @thomas11 

This resource is a little awkward, as generally (I think) this is about overriding a subscription-level setting. Setting `enabled`, `malwareScanning` or `sensitiveDataDiscovery` makes no sense without setting `overrideSubscriptionLevelSettings` to `true`:

![image](https://github.com/pulumi/pulumi-azure-native/assets/973909/2dd7d7a9-f6a2-48c4-8e6a-bb31be099379)

![image](https://github.com/pulumi/pulumi-azure-native/assets/973909/5513bf33-ffa6-4b96-9578-a4b5d56e0b97)

To put it another way, the "default" value cannot really be determined because it depends on the subscription's defaults. Is it sensible to therefore either a) do nothing on deletion but b) set overrideSubscriptionLevelSettings to `false`

**Comment by thomas11** [comment]
Hi @barclayadam, the latest Azure Native release v2.26.1 fixes this issue. Thank you for your patience.
